pkg/controller/controller_test.go

```diff
@@ -324,88 +328,127 @@ func TestResizePVC(t *testing.T) {
-			Name:             "Resize PVC with FS resize failure",
-			PVC:              createPVC(2, 1),
-			PV:               createPV(1, "testPVC", defaultNS, "foobar", &fsVolumeMode),
-			NodeResize:       true,
-			expansionFailure: true,
-			expectFailure:    true,
+			Name:       "Resize PVC with resourver version conflict should not emit event",
+			PVC:        createPVC(2, 1),
+			PV:         createPV(1, "testPVC", defaultNS, "foobar", &fsVolumeMode),
+			NodeResize: true,
+			expansionError: errors.NewConflict(schema.GroupResource{
+				Group:    v1.GroupName,
+				Resource: "persistentvolumeclaims",
+			}, "testPVC", fmt.Errorf("the object has been modified; "+
+				"please apply your changes to the latest version and try again")),
+			expectFailure: true,
+			expectedEvents: []string{
+				"Normal Resizing External resizer is resizing volume testPV",
+			},
 		},
 	} {
-		client := csi.NewMockClient("mock", test.NodeResize, true, false, true, true)
-		if test.expansionFailure {
-			client.SetExpansionFailed()
-		}
-		driverName, _ := client.GetDriverName(context.TODO())
+		t.Run(test.Name, func(t *testing.T) {
+			client := csi.NewMockClient("mock", test.NodeResize, true, false, true, true)
+			if test.expansionError != nil {
+				client.SetExpansionError(test.expansionError)
+			}
+			driverName, _ := client.GetDriverName(context.TODO())
 
-		var expectedCap resource.Quantity
-		initialObjects := []runtime.Object{}
-		if test.PVC != nil {
-			initialObjects = append(initialObjects, test.PVC)
-			expectedCap = test.PVC.Status.Capacity[v1.ResourceStorage]
-		}
-		if test.PV != nil {
-			test.PV.Spec.PersistentVolumeSource.CSI.Driver = driverName
-			initialObjects = append(initialObjects, test.PV)
-		}
+			var expectedCap resource.Quantity
+			var initialObjects []runtime.Object
+			if test.PVC != nil {
+				initialObjects = append(initialObjects, test.PVC)
+				expectedCap = test.PVC.Status.Capacity[v1.ResourceStorage]
+			}
+			if test.PV != nil {
+				test.PV.Spec.PersistentVolumeSource.CSI.Driver = driverName
+				initialObjects = append(initialObjects, test.PV)
+			}
 
-		kubeClient, informerFactory := fakeK8s(initialObjects)
-		pvInformer := informerFactory.Core().V1().PersistentVolumes()
-		pvcInformer := informerFactory.Core().V1().PersistentVolumeClaims()
-		podInformer := informerFactory.Core().V1().Pods()
+			kubeClient, informerFactory := fakeK8s(initialObjects)
+			pvInformer := informerFactory.Core().V1().PersistentVolumes()
+			pvcInformer := informerFactory.Core().V1().PersistentVolumeClaims()
+			podInformer := informerFactory.Core().V1().Pods()
 
-		csiResizer, err := resizer.NewResizerFromClient(client, 15*time.Second, kubeClient, driverName)
-		if err != nil {
-			t.Fatalf("Test %s: Unable to create resizer: %v", test.Name, err)
-		}
+			csiResizer, err := resizer.NewResizerFromClient(client, 15*time.Second, kubeClient, driverName)
+			if err != nil {
+				t.Fatalf("Test %s: Unable to create resizer: %v", test.Name, err)
```

Comment:
With `t.Run` above, it's no longer necessary to log the test name. This applies to all errors below.